src/components/cart/CartDetails.jsx

```diff
@@ -0,0 +1,20 @@
+import { useStateContext } from "../../context";
+import { ProductDeliverAddress } from "./ProductDeliverAddress";
+import { CartProducts } from "./CartProducts";
+
+export function CartDetails() {
+  const {
+    state: { productsInCart },
```

Comment:
Interesting implementation of destructuring done here for storing all context within one useContext